mesonbuild/backend/ninjabackend.py

```diff
@@ -3552,9 +3552,8 @@ def get_build_rpath_args(self, target: build.BuildTarget, linker: T.Union[Compil
             linker.build_rpath_args(self.environment,
                                     self.environment.get_build_dir(),
                                     target_slashname_workaround_dir,
-                                    self.determine_rpath_dirs(target),
-                                    target.build_rpath,
-                                    target.install_rpath))
+                                    self,
+                                    target))
```

Comment:
Instead of passing the entire backend object into the linker, maybe this should be a linker.uses_rpath_args property to avoid layering issues...